### PR TITLE
[Bug] Create dynamic partition table failed with enable create_history_partition and not specify the start value

### LINKED ISSUE #5995: [bug] Create dynamic partition table failed with enable create_history_partition and not specify the start value
**Describe the bug**
Create dynamic partiiton table failed, and getting a lot of invalid format date value message
enable the property "dynamic_partition.create_history_partition" and not set property "dynamic_partition.start"

2021-06-08 12:19:21,255 WARN (doris-mysql-nio-pool-0|147) [DynamicPartitionScheduler.getAddPartitionClause():169] Error in gen addPartitionKeyRange. Error=errCode = 2, detailMessage = date literal [+178951742-12-01] is invalid, db: default_cluster:poc, table: dwd_tf_order2

**To Reproduce**
create table ddl: 
create table poc.dwd_tf_order2 (
 `pt_date` date NULL comment "分区字段" ,
`order_id` varchar(255) NULL COMMENT "订单ID",
....
) ENGINE = OLAP
UNIQUE KEY(pt_date, `order_id`)
COMMENT "订单明细表"
PARTITION BY RANGE (pt_date)()
DISTRIBUTED BY HASH(`order_id`) BUCKETS 6
PROPERTIES (
"replication_num" = "1",
"in_memory" = "false",
"storage_format" = "V2",
"dynamic_partition.enable" = "true",
"dynamic_partition.time_unit" = "MONTH",
"dynamic_partition.end" = "3",
"dynamic_partition.prefix" = "p",
"dynamic_partition.buckets" = "6",
"dynamic_partition.create_history_partition" = "true");

### PR DESCRIPTION
**Pull request by harveyyue**
fix the issue #5995
Add the property "dynamic_partition.history_partition_num" to specify the history partition number when enable create_history_partition to fix the invalid date format value
and add these two properties to docs

## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have created an issue on (Fix #5995) and described the bug/feature there in detail
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/common/util/DynamicPartitionUtil.java [hunk 1]
```diff
@@ -348,13 +363,32 @@ public static void registerOrRemoveDynamicPartitionTable(long dbId, OlapTable ol
...
+        }
+
         // Check the number of dynamic partitions that need to be created to avoid creating too many partitions at once.
         // If create_history_partition is false, history partition is not considered.
+        int expect_create_partition_num;
         if (!createHistoryPartition) {
             start = 0;
+            expect_create_partition_num = end - start;
+        } else {
+            if (historyPartitionNum != 0) {
```
Comment (qidaye) [inline fe/fe-core/src/main/java/org/apache/doris/common/util/DynamicPartitionUtil.java:381]: Check `NOT_SET_HISTORY_PARTITION_NUM`

File: docs/en/sql-reference/sql-statements/Data Definition/CREATE TABLE.md [hunk 2]
```diff
@@ -312,6 +312,10 @@ Syntax:
        Dynamic_partition. Prefix: used to specify the partition name prefix to be created, such as the partition name prefix p, automatically creates the partition name p20200108
        
        Dynamic_partition. Buckets: specifies the number of partition buckets that are automatically created
+
```
Comment (qidaye) [inline docs/en/sql-reference/sql-statements/Data Definition/CREATE TABLE.md:315]: Consistent with Chinese document format.

File: fe/fe-core/src/main/java/org/apache/doris/common/util/DynamicPartitionUtil.java [hunk 3]
```diff
@@ -348,13 +363,32 @@ public static void registerOrRemoveDynamicPartitionTable(long dbId, OlapTable ol
             analyzedProperties.put(DynamicPartitionProperty.CREATE_HISTORY_PARTITION, val);
         }
 
+        int historyPartitionNum = 0;
```
Comment (qidaye) [inline fe/fe-core/src/main/java/org/apache/doris/common/util/DynamicPartitionUtil.java:366]: It is better to keep the same style as the existing code. 
You can refer to `REPLICATION_NUM` section.

**Comment by qidaye** [comment]
By the way, you also need to update dynamic partition doc https://github.com/apache/incubator-doris/blob/master/docs/en/administrator-guide/dynamic-partition.md, for both Chinese and English.

**Comment by harveyyue** [comment]
> By the way, you also need to update dynamic partition doc https://github.com/apache/incubator-doris/blob/master/docs/en/administrator-guide/dynamic-partition.md, for both Chinese and English.

Ok, I have done with these code review issue

File: fe/fe-core/src/main/java/org/apache/doris/clone/DynamicPartitionScheduler.java [hunk 4]
```diff
@@ -63,6 +63,8 @@
 import java.util.Map;
 import java.util.Set;
 
+import static org.apache.doris.catalog.DynamicPartitionProperty.NOT_SET_HISTORY_PARTITION_NUM;
```
Comment (qidaye) [inline fe/fe-core/src/main/java/org/apache/doris/clone/DynamicPartitionScheduler.java:66]: There is no need to import the static value. You can use it directly. `DynamicPartitionProperty.NOT_SET_HISTORY_PARTITION_NUM`

File: fe/fe-core/src/main/java/org/apache/doris/common/util/DynamicPartitionUtil.java [hunk 5]
```diff
@@ -363,32 +362,33 @@ public static void registerOrRemoveDynamicPartitionTable(long dbId, OlapTable ol
...
-            historyPartitionNum = checkHistoryPartitionNum(val);
+            checkHistoryPartitionNum(val);
             properties.remove(DynamicPartitionProperty.HISTORY_PARTITION_NUM);
             analyzedProperties.put(DynamicPartitionProperty.HISTORY_PARTITION_NUM, val);
         }
 
         // Check the number of dynamic partitions that need to be created to avoid creating too many partitions at once.
         // If create_history_partition is false, history partition is not considered.
-        int expect_create_partition_num;
+        int expectCreatePartitionNum;
```
Comment (qidaye) [inline fe/fe-core/src/main/java/org/apache/doris/common/util/DynamicPartitionUtil.java:374]: You can add some logic explaination here.

File: fe/fe-core/src/main/java/org/apache/doris/common/util/DynamicPartitionUtil.java [hunk 6]
```diff
@@ -363,32 +362,33 @@ public static void registerOrRemoveDynamicPartitionTable(long dbId, OlapTable ol
...
         if (!createHistoryPartition) {
             start = 0;
-            expect_create_partition_num = end - start;
+            expectCreatePartitionNum = end - start;
         } else {
-            if (historyPartitionNum != 0) {
-                expect_create_partition_num = end - Math.max(start, -historyPartitionNum);
+            int historyPartitionNum = Integer.valueOf(analyzedProperties.getOrDefault(DynamicPartitionProperty.HISTORY_PARTITION_NUM,
+                    String.valueOf(DynamicPartitionProperty.NOT_SET_HISTORY_PARTITION_NUM)));
+            if (historyPartitionNum > 0) {
```
Comment (qidaye) [inline fe/fe-core/src/main/java/org/apache/doris/common/util/DynamicPartitionUtil.java:381]: `historyPartitionNum != DynamicPartitionProperty.NOT_SET_HISTORY_PARTITION_NUM`

**Review by qidaye — COMMENTED** [review]
Thands for your hard work. Left few comments. 
Please add some unit tests for your changes.

File: fe/fe-core/src/main/java/org/apache/doris/clone/DynamicPartitionScheduler.java [hunk 7]
```diff
@@ -143,9 +145,19 @@ public void createOrUpdateRuntimeInfo(long tableId, String key, String value) {
         ZonedDateTime now = ZonedDateTime.now(dynamicPartitionProperty.getTimeZone().toZoneId());
 
         boolean createHistoryPartition = dynamicPartitionProperty.isCreateHistoryPartition();
-        int idx = createHistoryPartition ? dynamicPartitionProperty.getStart() : 0;
+        int idx;
```
Comment (qidaye) [inline fe/fe-core/src/main/java/org/apache/doris/clone/DynamicPartitionScheduler.java:148]: Some logical explanations can also be added here.

**Comment by qidaye** [comment]
LGTM

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**